src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/upgrade/upgrade-start-modal-component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, OnInit } from '@angular/core';
+import { Validators } from '@angular/forms';
+import { Observable, of } from 'rxjs';
+import { catchError, ignoreElements } from 'rxjs/operators';
+
+import { Icons } from '~/app/shared/enum/icons.enum';
+import { Permission } from '~/app/shared/models/permissions';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { UpgradeService } from '~/app/shared/api/upgrade.service';
+import { UpgradeInfoInterface } from '~/app/shared/models/upgrade.interface';
+import { CdFormBuilder } from '~/app/shared/forms/cd-form-builder';
+import { NotificationType } from '~/app/shared/enum/notification-type.enum';
+import { CdFormGroup } from '~/app/shared/forms/cd-form-group';
+import { NotificationService } from '~/app/shared/services/notification.service';
+
+@Component({
+  selector: 'cd-upgrade-start-modal-component',
+  templateUrl: './upgrade-start-modal-component.html',
+  styleUrls: ['./upgrade-start-modal-component.scss']
+})
+export class UpgradeStartModalComponent implements OnInit {
+  permission: Permission;
+  upgradeInfoError$: Observable<any>;
+  upgradeInfo$: Observable<UpgradeInfoInterface>;
+  upgradeForm: CdFormGroup;
+  icons = Icons;
+
+  constructor(
+    public actionLabels: ActionLabelsI18n,
+    private authStorageService: AuthStorageService,
+    private fb: CdFormBuilder,
+    public activeModal: NgbActiveModal,
+    private upgradeService: UpgradeService,
+    private notificationService: NotificationService
+  ) {
+    this.permission = this.authStorageService.getPermissions().configOpt;
+  }
+
+  ngOnInit() {
+    this.upgradeInfo$ = this.upgradeService.list();
+    this.upgradeInfoError$ = this.upgradeInfo$?.pipe(
```

Comment:
we don't need to do this again. We can pass the versions from the upgrade.component where we are actually doing the subscription. so on this file you can introduce a variable called `versions` to catch the params that are send from upgrade component.